src/OddJobs/Job.hs

```diff
@@ -573,34 +573,34 @@ jobEventListener = do
           [Only (_ :: JobId)] -> pure $ Just jid
           x -> error $ "WTF just happned? Was expecting a single row to be returned, received " ++ (show x)
 
-  withResource pool $ \monitorDbConn -> do
-    void $ liftIO $ PGS.execute monitorDbConn ("LISTEN ?") (Only $ pgEventName tname)
+  withRunInIO $ \runInIO -> withResource pool $ \monitorDbConn -> do
+    void $ PGS.execute monitorDbConn ("LISTEN ?") (Only $ pgEventName tname)
     forever $ do
-      log LevelDebug $ LogText "[LISTEN/NOTIFY] Event loop"
-      notif <- liftIO $ getNotification monitorDbConn
-      concurrencyControlFn >>= \case
-        False -> log LevelWarn $ LogText "Received job event, but ignoring it due to concurrency control"
+      runInIO $ log LevelDebug $ LogText "[LISTEN/NOTIFY] Event loop"
```

Comment:
I guess `runInIO` is required everywhere due to the top-level `withRunInIO`. Is there a "lifted" version of the data pool library that we can use instead? Or write our own wrapper?

Reply:
withResource is now an IO block: https://hackage.haskell.org/package/resource-pool-0.3.1.0/docs/Data-Pool.html#v:withResource

we need to unlift to get access to the logger and everything else from the polymorhpic mtl stack.
I'm not sure what type you want to give the wrapper.